packages/lodestar-cli/src/cmds/beacon/cmds/run/run.ts

```diff
@@ -24,28 +25,26 @@ export async function run(options: Arguments<IBeaconArgs & Partial<IBeaconNodeOp
   await initBLS();
 
   // Auto-setup testnet
-  if (options.testnet) {
-    await initBeacon(options);
-  }
+  if (options.testnet) await initBeacon(options);
 
   options = deepmerge(defaultOptions, options) as Arguments<IBeaconArgs & Partial<IBeaconNodeOptions>>;
-
+  const logger = new WinstonLogger();
   const peerId = await readPeerId(options.network.peerIdFile);
   // read local enr from disk
-  options.network.discv5.enr = await readEnr(options.network.enrFile);
+  const enr = await readEnr(options.network.enrFile);
+  if (enr.multiaddrUDP && !isLocalMultiAddr(enr.multiaddrUDP)) {
+    logger.info(`MultiAddress of ${enr.multiaddrUDP.toString()} is not part of local network interface so removed it`);
+    clearMultiaddrUDP(enr);
+  }
+  options.network.discv5.enr = enr;
```

Comment:
How come we don't handle this in network module?

Reply:
I think it's just more clear to put it here because we want to handle this logic at startup time.
what do you think @wemeetagain ?

Reply:
This fix changes should not apply to the validator when it's run by a different consumer than our CLI?

Reply:
I'm not sure we want the same behavior with a browser node, at least, not the _exact_ same behavior.
(eg: we might want to explicitly set a relay IP, or have a more intricate ENR initialization procedure in that case)

I think we can leave this here for now, or maybe move it to `createNodeJsLibp2p`?

Reply:
I guess a bigger issue is figuring out a strategy for splitting network agnostic code from nodejs / browser specific code. We have done some work towards that end, eg: with a nodejs-specific libp2p bundle, but it there likely will be a need for better organization.